### PR TITLE
WIP: MAD-29-News-API-routes

### PR DESCRIPTION
**Pull request by NoahHakansson**

### CODE REVIEW

**Comment by NoahHakansson** [comment]
eyy nice job! just thinking that the python script should be in `backend/api/scripts` and not in `backend/api/src` because the `src` directory is for go files. to keep everything nicely seperated.

**Comment by NoahHakansson** [comment]
test comment fom neovim

**Comment by NoahHakansson** [comment]
Merge main into this branch by running `git checkout main` > `got pull` > `git checkout MAD-29-News-API-routes` > `git merge main`

**Comment by NoahHakansson** [comment]
This needs fixing `main.go` firstly has old code so after merging check how other things are called in `main.go` with the `go` keyword to create a go-routine(thread). 

The news scraper script should run maybe every 5 hours in a loop; Scraping and after scraping make sure that the news that got scraped are added to the database. Since news in the database have Title as a unique key, we CAN just add duplicates, but maybe there is a better way in GORM to avoid duplicates than just adding them and letting the database handle it with an error. Maybe BATCH INSERT in GORM would be an option? (lets you can add many things at once, with an array/slice). Just optional tho if you want to, its fine to just loop and call your `CreateNews()` function for each news in the list of news from the scraper.

**Comment by NoahHakansson** [comment]
Might be a decent amount of merge conflicts since we have done a few big changes since main was merged into this branch. So just double check extra and ask if you wonder about any merge conflicts what should be kept or removed etc.

File: backend/api/main.go [hunk 1]
```diff
@@ -17,11 +18,15 @@ func main() {
 		log.Println("DEV: Could not load .env file")
 	}
 
+	// routes.SetupRoutes()
```
Comment (NoahHakansson) [inline backend/api/main.go:21]: Remove unused comment.

File: backend/api/src/Scraper/scraper.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+		GetNews(gormDB)
+		time.Sleep(5 * time.Hour)
+	}
+}
+
+// GetNews function to get news from the website
+func GetNews(gormDB *gorm.DB) {
+	// Instantiate default collector
+	c := colly.NewCollector(
+		colly.AllowedDomains("www.bth.se"),
```
Comment (NoahHakansson) [inline backend/api/src/Scraper/scraper.go:26]:
```suggestion
		colly.AllowedDomains("www.bth.se"),
Maybe move to a constant variable at the top of file.```

File: backend/api/src/Scraper/scraper.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			}
+
+			err = db.CreateNews(gormDB, &article) // Create the news in the database
+			if err != nil {
+				log.Println(err)
+			}
+		})
+	})
+
+	err := c.Visit("https://www.bth.se/category/nyheter")
```
Comment (NoahHakansson) [inline backend/api/src/Scraper/scraper.go:80]: Move to a constant at the top of the file aswell maube?

File: backend/api/main.go [hunk 4]
```diff
@@ -4,9 +4,10 @@ package main
 import (
 	"log"
 
-	"github.com/abodsakah/BTH-appen/backend/api/src/DB"
-	"github.com/abodsakah/BTH-appen/backend/api/src/Notifications"
-	"github.com/abodsakah/BTH-appen/backend/api/src/Routes"
+	db "github.com/abodsakah/BTH-appen/backend/api/src/DB"
+	notifications "github.com/abodsakah/BTH-appen/backend/api/src/Notifications"
```
Comment (NoahHakansson) [inline backend/api/main.go:8]:
```suggestion
	"github.com/abodsakah/BTH-appen/backend/api/src/Notifications"
```

File: backend/api/main.go [hunk 5]
```diff
@@ -4,9 +4,10 @@ package main
...
+	routes "github.com/abodsakah/BTH-appen/backend/api/src/Routes"
```
Comment (NoahHakansson) [inline backend/api/main.go:9]:
```suggestion
	"github.com/abodsakah/BTH-appen/backend/api/src/Routes"
```

File: backend/api/main.go [hunk 6]
```diff
@@ -4,9 +4,10 @@ package main
...
+	scraper "github.com/abodsakah/BTH-appen/backend/api/src/Scraper"
```
Comment (NoahHakansson) [inline backend/api/main.go:10]:
```suggestion
	"github.com/abodsakah/BTH-appen/backend/api/src/Scraper"
```

File: backend/api/src/Scraper/scraper.go [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+	// Instantiate default collector
+	c := colly.NewCollector(
+		colly.AllowedDomains("www.bth.se"),
+	)
+
+	c.OnHTML(".Article-result", func(h *colly.HTMLElement) {
+		// Get the title
+		h.ForEach(".ArticleItem", func(_ int, e *colly.HTMLElement) {
+			title := e.ChildText("h2")
+			fmt.Println(title)
```
Comment (NoahHakansson) [inline backend/api/src/Scraper/scraper.go:33]: Remove debug prints.

File: backend/api/src/Scraper/scraper.go [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+				childNodes := selction.Children().Nodes
+				if len(childNodes) > 1 {
+					des = selction.FindNodes(childNodes[1]).Text()
+				} else if len(childNodes) == 1 {
+					des = selction.FindNodes(childNodes[0]).Text()
+				}
+			})
+			// Get the link
+			link := e.ChildAttr("a", "href")
+			fmt.Println(link)
```
Comment (NoahHakansson) [inline backend/api/src/Scraper/scraper.go:64]: Remove debug prints.

File: backend/api/main.go (hunk 6 above)
Comment (najemhamo) [reply backend/api/main.go:10]: I couldn't delete them, I'm getting them back as soon as I save the file!

**Comment by NoahHakansson** [comment]
Looks good, Gabi can do a review as well :+1:

**Pull request merged**